### DISCUSSION TITLE
Can duckdb CLI use environment variable values in SET statements?

### DISCUSSION THREAD

**Discussion by mskyttner** [duckdb/duckdb#3522] [Q&A] [answered]
An example of a use case is (containerized or non-containerized) duckdb CLI using a SET (or PRAGMA) statement where for example the number of threads or the connection information for an S3 backend serving parquet files is something that is dynamically determined on the machine running duckdb CLI and also available as environment variable values.

Is there a way in duckdb CLI to run a SET statement or PRAGMA statement with values coming from such an environment variable?

For example, consider an init.sql script (run either with `duckdb -init init.sql` or in the duckdb shell using `.read init.sql`) with contents for example like this:

```
install 'httpfs';
load 'httpfs';
set s3_endpoint=$MINIO_ENDPOINT;
set s3_access_key_id=$MINIO_USER;
set s3_secret_access_key=$MINIO_PASS;
.timer on
select count(*) from 's3://$MINIO_BUCKET/*.parquet';
.shell echo "Number of CPUs: $(nproc --all)"
```

**Comment by Mytherin** [comment, marked as answer]
This is not possible right now, but we could probably envision adding a `getenv` function to the CLI.
